src/main/scala/uk/gov/nationalarchives/omega/api/services/Dispatcher.scala

```diff
@@ -41,16 +41,27 @@ class Dispatcher(val localProducer: LocalProducer, localMessageStore: LocalMessa
 
   def run(dispatcherId: Int)(q: Queue[IO, LocalMessage]): IO[Unit] =
     for {
-      requestMessage          <- q.take
-      validatedRequestMessage <- IO.pure(validateLocalMessage(requestMessage))
-      _ <- logger.info(s"Dispatcher # $dispatcherId, processing message id: ${requestMessage.persistentMessageId}")
-      (businessService, businessServiceRequest) <- IO.pure(createServiceRequest(validatedRequestMessage))
-      validatedBusinessServiceRequest <-
-        IO.pure(validateBusinessServiceRequest(businessService, businessServiceRequest))
-      businessResult <- execBusinessService(businessService, validatedBusinessServiceRequest)
-      res            <- sendResultToJmsQueue(businessResult, validatedRequestMessage)
-      _              <- remove(requestMessage)
-    } yield res
+      localMessage <- q.take
+      _ <- logger.info(s"Dispatcher # $dispatcherId, processing message id: ${localMessage.persistentMessageId}")
+      _ <- process(localMessage)
+      _ <- remove(localMessage)
+    } yield ()
+
+  private def process(localMessage: LocalMessage): IO[Unit] =
+    localMessage.validate() match {
+      case Validated.Valid(validatedLocalMessage: ValidatedLocalMessage) =>
+        val (businessService: BusinessService, businessServiceRequest: BusinessServiceRequest) =
+          createServiceRequest(validatedLocalMessage)
+        validateBusinessServiceRequest(businessService, businessServiceRequest) match {
+          case Validated.Valid(validatedBusinessServiceRequest) =>
+            sendResultToJmsQueue(
+              execBusinessService(businessService, validatedBusinessServiceRequest),
+              validatedLocalMessage
+            )
+          case Validated.Invalid(errors) => localProducer.sendWhenBusinessRequestIsInvalid(localMessage, errors)
```

Comment:
The nested case match could be handled in a separate function to make it more readable.